### ISSUE TITLE
`isinstance` detects module as a type

### ISSUE THREAD

**Issue by AlanBogarin** [microsoft/pyright#10605]
**Describe the bug**
No warning is displayed about the misuse of `isinstance` when an argument is not a type,

**Code or Screenshots**
```python
import datetime
import typing

a = typing.cast(datetime.datetime | None, None)

isinstance(a, datetime)  # error: datetime is not a type
```

```
>>> py -m pyright test.py
0 errors, 0 warnings, 0 informations
```

**VS Code extension or command-line**
I'm using Pylance in strict mode and this issue arose. When I tested it on the command line, it persisted.

Pylance: 2025.6.1
Pyright (pip): 1.1.402

**Comment by erictraut** [comment]
Thanks, I agree this is a bug. It's more general than `isinstance` though.

```python
import datetime
x: type = datetime # Should be an error
```
